packages/internals/src/getPackedPackage.ts

```diff
@@ -36,60 +36,21 @@ export async function getPackedPackage(name: string, target?: string, packageDir
   }
 
   if (!packageDir) {
-    throw new Error(`Error in getPackage: Could not resolve package ${name} from ${__dirname}`)
+    throw new Error(`Error in getPackage: Could not resolve package ${name} from ${process.cwd()} target ${target}`)
   }
-  const tmpDir = tempy.directory() // thanks Sindre
-  const archivePath = path.join(tmpDir, `package.tgz`)
 
-  // Check if yarn is available.
-  const isYarn = await hasYarn(packageDir)
+  const tmpDir = target ?? tempy.directory() // thanks Sindre
 
-  const packCmd = isYarn ? ['yarn', 'pack', '-f', archivePath] : ['npm', 'pack', packageDir]
+  const pkgJson = require(`${packageDir}/package.json`)
 
-  // shell-quote args
-  const escapedCmd = quote(packCmd)
+  // we compute the paths of the files that would get npm published
+  const pkgFiles = (pkgJson.files ?? []).concat(['package.json']).map((file: string) => path.join(packageDir!, file))
 
-  // pack into a .tgz in a tmp dir
-  await execa.command(escapedCmd, {
-    shell: true,
-    cwd: isYarn ? packageDir : tmpDir, // for npm pack it outputs a file to the cwd
-  })
-
-  if (!isYarn) {
-    // since npm pack does not have option to specify a filename we change it here
-    // find single tgz in temp folder
-    const filename = (await readdir(tmpDir))[0]
-    // rename it to match expected filename
-    await rename(path.join(tmpDir, filename), archivePath)
-  }
-
-  // extract and delete the archive
-  await tar.extract({
-    cwd: tmpDir,
-    file: archivePath,
-  })
-
-  await del(archivePath)
-
-  /**
-   * Only if a target is provided, perform the copy
-   */
-  if (target) {
-    // make target dir
-    await makeDir(target)
-
-    // copy stuff over
-    await copy({
-      from: path.join(tmpDir, 'package'), // when using yarn pack and extracting it, it includes a folder called "package"
-      to: target,
-      recursive: true,
-      parallelJobs: 20,
-      overwrite: true,
-    })
-
-    // delete tmp dir
-    await del(tmpDir)
+  // we copy each file that we found in pkg to a new destination
+  for (const file of [...pkgFiles]) {
+    const dest = path.join(tmpDir, path.basename(file))
+    await copy(file, dest, { recursive: true, overwrite: true })
```

Comment:
Removing the packing via `yarn` which is terribly slow and caused memory overflows on CI.

Reply:
This will soon not be needed anymore anyways as the new test setup does things differently.